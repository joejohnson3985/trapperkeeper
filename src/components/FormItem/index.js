import React, { Component } from 'react';
import add from '../../media/icons/add.svg';
import box from '../../media/icons/check_box_blank.svg';
import checkedBox from '../../media/icons/check_box.svg';
import clear from '../../media/icons/clear.svg';
import PropTypes from 'prop-types';


class FormItem extends Component {
  constructor() {
    super();

    this.state = {
      value: '',
      checked: false
    }
  }

  componentDidMount() {
    this.populateForm()
  }

  populateForm = () => {
    if(this.props.item) {
      this.setState({ value: this.props.item, id: this.props.list_id, checked: this.state.checked})
    } else {
      this.setState({id: Date.now()})
    }
  }

  handleChange = (e) => {
    const { value } = e.target;
    this.setState({ value });
  }

  toggleCheckBox = () => {
    const checked = !this.state.checked;
    this.setState({ checked }, () => {
      this.props.handleItemSubmit({ list_id: this.state.id, item: this.state.value, checked: this.state.checked });
    });
  }

  handleKeypress = (e) => {
    if(e.key === 'Enter') {
      this.props.handleItemSubmit({ list_id: this.state.id, item: this.state.value, checked: this.state.checked });
    }
  }

  render() {
    const {checked, value} = this.state;
    const boxIcon = 
    <img 
      onClick={this.toggleCheckBox} 
      className='icon check-box' 
      role='checkbox' 
      aria-checked={checked}
      src={checked ? checkedBox : box} 
      alt='Checkbox' 
    />;
    const plus = <img className='icon add' src={add} alt='Add item' />
    const prevIcon = value.length ? boxIcon : plus;
    return (
      <div className='FormItem' id={this.props.list_id}>
        {prevIcon}
        <input 
          className={`item-text ${checked && 'checked-item'}`}
          type='text'
          onChange={this.handleChange}
          onKeyPress={this.handleKeypress}
          placeholder='List item'
          value={value}
          contentEditable={true}
        />
        <img onClick={()=> this.props.removeItem(this.props.list_id)} className='icon clear' src={clear} alt='Clear item' />
      </div>
    );  
  }
}

FormItem.propTypes = {
  checked: PropTypes.bool,
  handleItemSubmit: PropTypes.func,
  removeItem: PropTypes.func,
  list_id: PropTypes.number,
  item: PropTypes.string
};

export default FormItem;