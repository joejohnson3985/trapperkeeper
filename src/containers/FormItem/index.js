import React, { Component } from 'react';
import add from '../../media/icons/add.svg';
import box from '../../media/icons/check_box_blank.svg';
import checkedBox from '../../media/icons/check_box.svg';
import clear from '../../media/icons/clear.svg';

class FormItem extends Component {
  constructor() {
    super();

    this.state = {
      empty: true,
      value: '',
      checked: false
    }
  }

  componentDidMount() {
    this.populateForm()
  }

  populateForm = () => {
    if(this.props.item) {
      this.setState({ value: this.props.item, id: this.props.list_id})
    } else {
      this.setState({id: Date.now()})
    }
  }

  handleChange = (e) => {
    const { value } = e.target;
    this.setState({ value });
  }

  toggleCheckBox = (e) => {
    const checked = !this.state.checked;
    this.setState({ checked });
  }

  handleKeypress = (e) => {
    const idToUse = this.props.list_id
    if(e.key === 'Enter') {
    this.props.addFormItem({ list_id: this.state.id, item: this.state.value, checked: this.state.checked });
    this.setState({value: ''})
    }
  }
  render() {
    const {checked, value} = this.state;
    const boxIcon = 
    <img 
      onClick={this.toggleCheckBox} 
      className='icon check-box' 
      role='checkbox' 
      aria-checked={checked}
      checked={checked} 
      src={checked ? checkedBox : box} 
      alt='Checkbox' 
    />;
    const plus = <img className='icon add' src={add} alt='Add item' />
    const prevIcon = value.length ? boxIcon : plus;
    return (
      <div className='FormItem'>
        {prevIcon}
        <input 
          className='item-text'
          type='text'
          onChange={this.handleChange}
          onKeyPress={this.handleKeypress}
          placeholder='List item'
          value={value}
          contentEditable={true}
        />
        <img className='icon clear' src={clear} alt='Clear item' />
      </div>
    );  
  }
}

export default FormItem;