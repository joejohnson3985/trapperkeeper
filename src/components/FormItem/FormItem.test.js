import React from 'react';
import FormItem from './index';
import { shallow } from 'enzyme';

describe('FormItem', () => {
  let wrapper;
  let mockFunc = jest.fn()
  let mockItem = {item:'do the dishes'}

  beforeEach(() => {
    wrapper = shallow(
      <FormItem handleItemSubmit={mockFunc} 
                {...mockItem} 
                key={mockItem.list_id} />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      value:'do the dishes',
      checked: false
    })
  });
})

