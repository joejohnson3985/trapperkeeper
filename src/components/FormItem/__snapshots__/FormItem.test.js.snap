// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItem should match the snapshot 1`] = `
<div
  className="FormItem"
  id={1}
>
  <img
    alt="Checkbox"
    className="icon check-box"
    onClick={[Function]}
    role="checkbox"
    src="check_box_blank.svg"
  />
  <input
    className="item-text"
    contentEditable={true}
    onBlur={[Function]}
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="New List item"
    type="text"
    value="do the dishes"
  />
  <img
    alt="Clear item"
    className="icon clear"
    onClick={[Function]}
    src="clear.svg"
  />
</div>
`;
